discoverer/src/main/java/eu/seaclouds/platform/discoverer/core/Offering.java

```diff
@@ -171,10 +172,12 @@ public String toTosca() {
         if (this.toscaString != null)
             return this.toscaString;
 
-        /* otherwise, if there is the path of the file where the offering is stored, the file is read */
+        /* otherwise, if the path of the file is valid, then it is possible to read the offering from file */
         if (this.offeringPath != null) {
             try {
-                this.toscaString = new String(Files.readAllBytes(Paths.get(this.offeringPath)));
+                String offeringDirectory = Discoverer.instance().offeringManager.getOfferingDirectory().getAbsolutePath();
+                String offeringFileName = "offer_" + this.offeringId + ".yaml";
+                this.toscaString = new String(Files.readAllBytes(Paths.get(offeringDirectory + File.separatorChar + offeringFileName)));
                 return this.toscaString;
             } catch (IOException e) { }
```

Comment:
An IOException is usual here? If you think the program should not crash, at least log the exception. Otherwise you'll never know if something wrong is happening.